Pacifique78/MyDiary
# #169434240 Refactor back end code flow

#### What does this PR do?
Refactor back end code flow

#### Description of Task to be completed
>-separate routes
>-decrease if statements by implementing try and catch brokes
>-delete dummy data from modals
>-separate controllers
>-separate tests
>-add helpers
>-use await for checking if a user exists

#### How should this be manually tested?
>- clone the repo
>- run `npm install` and `npm run dev` to start the server
>- use either postman or any other testing tool to see if all endpoints are working 
>- run `npm test` to see if all the test are passing
>- Check if the Travis test are passing

#### What are the relevant pivotal tracker stories?
#169434240

#### Screenshots (if appropriate)
![Screenshot (83)](https://user-images.githubusercontent.com/50268181/67855975-3d714400-fb1c-11e9-92e0-bc989633c5ff.png)
![Screenshot (85)](https://user-images.githubusercontent.com/50268181/67855976-3d714400-fb1c-11e9-9e3a-fcfa275bae97.png)
![Screenshot (86)](https://user-images.githubusercontent.com/50268181/67855977-3e09da80-fb1c-11e9-81b3-eb6c2412f8ac.png)
![Screenshot (87)](https://user-images.githubusercontent.com/50268181/67855979-3ea27100-fb1c-11e9-8fb8-9a39cb27db5e.png)
![Screenshot (88)](https://user-images.githubusercontent.com/50268181/67855980-3ea27100-fb1c-11e9-85d5-7b797784d6c2.png)
